app/controllers/evidence_controller.rb

```diff
@@ -122,11 +122,12 @@ def destroy
           # 2. from the node evidence tab
           # 3. from the evidence show page itself (under node)
           # When using redirect_back in case 3, we find an evidence not found error,
-          # since the evidence does not exist anymore. That's why we check the 'Referer' here:
-          if request.headers['Referer'] == project_node_evidence_url(current_project, @node, @evidence)
-            redirect_to project_node_path(current_project, @node), notice: notice
+          # since the evidence does not exist anymore. That's why we check the referrer here.
+          referrer_path = URI(request.referrer).tap { |u| u.query = nil }.to_s
+          if !referrer_path || referrer_path == project_node_evidence_url(current_project, @node, @evidence)
+            redirect_to project_node_path(current_project, @node, tab: 'evidence-tab'), notice: notice
           else
-            redirect_back fallback_location: project_node_path(current_project, @node), notice: notice
+            redirect_to "#{referrer_path}?tab=evidence-tab", notice: notice
```

Comment:
I think I can summarize all my comments in this code proposition:
```
tab = { tab: 'evidence-tab' }
if !request.referrer || URI(request.referrer).path == project_node_evidence_path(current_project, @node, @evidence)
  redirect_to project_node_path(current_project, @node, tab), notice: notice
else
  query = (URI(request.referrer).query || '').chomp(tab.to_query) << tab.to_query
  redirect_to "#{URI(request.referrer).path}?#{query}", notice: notice
end
```

Ideas to improve it welcome to

Reply:
if we end up removing `redirect_back`, it would be nice to add a comment like
```
we would like to just redirect_back here
(https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back)
but that does not allow us to include the 'tab' param
```

Reply:
And one last idea: if we fine tune the UX so when we click the evidence tab on an issue, the url of the page really contain `?tab=evidence' , we could just leave the backend code almost as it was?
I've not explored that option.

Reply:
I think something like this should work:
```
referrer = URI(request.referrer || '')
tab = { tab: 'evidence-tab' }
if !request.referrer || referrer.path == project_node_evidence_path(current_project, @node, @evidence)
  redirect_to project_node_path(current_project, @node, tab), notice: notice
else
  referrer.query = tab.to_query
  redirect_to referrer, notice: notice
end
```
thoughts?